src/core/lib/channel/channelz.h

```diff
@@ -229,10 +229,45 @@ class ServerNode : public BaseNode {
   ChannelTrace trace_;
 };
 
+// helper class for holding and rendering the information about a particular
+// socket's address
+class SocketAddress {
```

Comment:
My only question is will there ever be any information that is NOT in the peer string, that the channelz object will need to determine the address type? Or will having the peer_string always be enough to determine TCP vs UDS vs future other address type

I do agree though, the code in chttp2 has become too long... I will take a shot at the refactor.

Reply:
I can't think of any case where it will need info that is not in the peer string.  But even if it does, if the API is just that the transport passes in a string, the transport can construct that string to contain whatever information it needs.  The only thing that the channelz code needs to know is that if it doesn't parse as an IP or unix address, then it should return the string verbatim as an `OtherAddress`.

Reply:
Ack, I will implement this today.